gui/qt/amountedit.py

```diff
@@ -93,7 +93,7 @@ def get_amount(self):
         except:
             return None
         p = pow(10, self.decimal_point())
-        return int( p * x ) if x > 0 else None
+        return int( p * x )
```

Comment:
Is this safe or desired?  Why remove this check? I tested it and the amountedit behaves essentially the same but-- perhaps this would have unintended consequences?

Just let me know, @jhoenicke  that you looked at the code and are sure it's ok.

Reply:
The only obvious consequence is it allows you to send 0 coins.  Which isn't a bad thing.  You can consolidate funds and I presume you did this to allow for OP_RETURN tx's?

Reply:
This is so that the user can explicitly set the amount to 0, which is necessary for OP_RETURN.  Otherwise electron-cash shows an error if the amount is None and refuses to continue.

One could make more checks, e.g. if an output that is not OP_RETURN is zero, or that the amount is not below the dust limit (is dust limit still in bitcoin cash?).  But I wanted to keep the patch small, to just add OP_RETURN functionality.

Reply:
Ok .. I figured as much.  Took me a second to figure out why.  

Ok well I'll let this "sit" today for a few hours then I'll merge it.  I really like it and it killes 2 issues and works great.  I'm really glad you submitted this man! Thanks!